bluephos/tasks/dft_orca.py

```diff
@@ -0,0 +1,141 @@
+END
+* xyz 0 3 {0}
+"""
+
+TRIPLET_ORCA_INPUT = """!B3LYP LANL2DZ
+%PAL NPROCS 4 END
+* xyzfile 0 3 {0}
+"""
+
+BASE_ORCA_INPUT = """!B3LYP LANL2DZ
+%PAL NPROCS 4 END
+* xyzfile 0 1 {0}
+"""
+
+
+def create_orca_input_files(temp_dir, xyz_value):
+    """Generate ORCA input files based on the template and provided molecular structure."""
+
+    relax_input_file = os.path.join(temp_dir, "relax.inp")
+    triplet_input_file = os.path.join(temp_dir, "triplet.inp")
+    base_input_file = os.path.join(temp_dir, "base.inp")
+    relax_xyz_file = os.path.join(temp_dir, "relax.xyz")
+
+    with open(relax_input_file, "w") as file:
+        file.write(ORCA_INPUT_OPT.format(xyz_value))
+
+    with open(triplet_input_file, "w") as file:
+        file.write(TRIPLET_ORCA_INPUT.format(relax_xyz_file))
+
+    with open(base_input_file, "w") as file:
+        file.write(BASE_ORCA_INPUT.format(relax_xyz_file))
+
+    return relax_input_file, triplet_input_file, base_input_file
+
+
+def run_orca_command(input_file, output_file, orca_path):
+    """Execute the ORCA software command using subprocess to perform quantum chemical calculations."""
+
+    command = [orca_path, input_file]
+    with open(output_file, "w") as output:
+        subprocess.run(command, stdout=output, stderr=subprocess.PIPE, check=True, text=True)
+
+
+def run_dft_calculation(temp_dir, orca_path, xyz_value, base_name):
+    """Run DFT calculations for the specified molecule using ORCA."""
+
+    relax_input, triplet_input, base_input = create_orca_input_files(temp_dir, xyz_value)
+    relax_output = os.path.join(temp_dir, f"{base_name}_relax_output.txt")
+    triplet_output = os.path.join(temp_dir, f"{base_name}_triplet_output.txt")
+    base_output = os.path.join(temp_dir, f"{base_name}_base_output.txt")
+
+    try:
+        run_orca_command(relax_input, relax_output, orca_path)
+        run_orca_command(triplet_input, triplet_output, orca_path)
+        run_orca_command(base_input, base_output, orca_path)
+    except subprocess.CalledProcessError as e:
+        logging.error(f"DFT calculation failed: {e}")
+        raise
+
+    return triplet_output, base_output
+
+
+def remove_second_row(xyz):
+    """Remove the second row from the xyz data to comply with ORCA input requirements."""
+
+    lines = xyz.split("\n")
+    if len(lines) > 1:
+        lines.pop(1)
+    lines.append("*")  # Add a new line with '*'
+    return "\n".join(lines)
+
+
+def process_dataframe(df: pd.DataFrame) -> pd.DataFrame:
+    """Process each valid entry in the DataFrame for DFT calculations."""
+
+    orca_path = os.path.join(os.getenv("EBROOTORCA"), "orca")
+
+    for index, row in df.iterrows():
+
+        base_name = None
+        if row["xyz"] not in ["failed", None]:
+            base_name = row["ligand_identifier"]
+
+        if base_name:
+            if DEBUG:
+                temp_dir = tempfile.mkdtemp()
+            else:
+                temp_dir = tempfile.TemporaryDirectory()
+
```

Comment:
Yes, that's an elegant way to manage temporary files/directories and we can remove try... block here right now since we're not currently handling specific errors

Reply:
But however, looks like in the Python3.11 (that's version on my project) the tempfile.TemporaryDirectory class does not support a delete parameter in its constructor.

Reply:
ah shucks! I must have been playing with 3.12 recently without realizing it. It might not be good for this project to depend on 3.12+ - so either can leave it or create our own such context manager

Reply:
OK, So I just keep this currently